target/linux/ath79/dts/ar9344_netgear_wndr.dtsi

```diff
@@ -171,22 +171,25 @@
 			reg = <0x3c0000 0x300000>;
 		};
 
-		kernel@6c0000 {
-			label = "kernel";
-			reg = <0x6c0000 0x400000>;
-		};
-
-		ubiconcat0: partition@ac0000 {
-			label = "ubiconcat0";
-			reg = <0xac0000 0x1500000>;
-		};
-
 		partition@6c0000 {
 			label = "firmware";
 			reg = <0x6c0000 0x1900000>;
-			compatible = "openwrt,uimage", "denx,uimage";
+			compatible = "openwrt,uimage", "denx,uimage", "fixed-partitions";
 			openwrt,ih-magic = <0x33373033>;
 			openwrt,ih-type = <IH_TYPE_FILESYSTEM>;
```

Comment:
These uimage stuffs are used to split the rootfs and the kernel partitions. You can just drop them as the kernel partition address has already been manually defined.

Reply:
Oh so fixed-partitions is a replacement for uimage?

Reply:
huh interesting. the only other usage of IH_TYPE_FILESYSTEM seems to be netgear devices.

Reply:
> Oh so fixed-partitions is a replacement for uimage?

Yes.

> huh interesting. the only other usage of IH_TYPE_FILESYSTEM seems to be netgear devices.

Those old devices have special fake squashfs3 filesystem with built-in kernel. It's used to support this format.

To be clear, this error was firstly introduced by the wrong `compatible = "netgear,uimage";` in the device support commit https://github.com/openwrt/openwrt/commit/9b6f89c37fc9c2e23cc123bd7b5cf9a1f8dc301a. ath79 NAND devices don't need this property as they have standalone kernel partition.